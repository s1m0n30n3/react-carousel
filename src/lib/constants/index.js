/**
 * @enum string
 */
export const GalleryMode = {
  BOUNDED: "bounded",
  RETURN: "return",
};

/**
 * @enum string
 */
export const TransitionEffect = {
  SLIDE: "slide",
  FADE: "fade",
  IN_OUT_SLIDE: "inOutSlide",
};

/**
 * @enum string
 */
export const MouseEvent = {
  END: "end",
  START: "start",
  MOVE: "move",
};

/**
 * @enum string
 */
export const CarouselAction = {
  PREV: "prev",
  NEXT: "next",
};

/**
 * @enum string
 */
export const ImageBehavior = {
  FULL: "full",
  CONTAINED: "contained",
};

/**
 * @enum number[]
 */
export const GalleryRatio = {
  R1_1: [1, 1],
  R4_1: [4, 1],
  R3_2: [3, 2],
  R4_3: [4, 3],
  R5_4: [5, 4],
  R16_10: [16, 10],
  R16_9: [16, 9],
  R21_9: [21, 9],
  R32_9: [32, 9],
};

/**
 * @enum string
 */
export const NavigationType = {
  ONLY_ARROWS: "onlyArrows",
  ONLY_DOTS: "onlyDots",
  ARROWS_AND_DOTS: "arrowsAndDots",
  NONE: "none",
};

/**
 * @enum string
 */
export const NavigationPosition = {
  TOP: "top",
  CENTER: "center",
  BOTTOM: "bottom",
  BOTTOM_OUTSIDE: "bottomOutside",
};
