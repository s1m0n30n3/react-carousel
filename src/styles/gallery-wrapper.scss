.gallery-wrapper--base {
    height: inherit;
    transition-timing-function: ease-in-out;
    display: flex;
}

.gallery-wrapper--slide-effect {
    transition-property: transform;
}

.gallery-wrapper--fade-effect {
    width: 100%;
    transition-property: opacity;
    position: relative;
}
