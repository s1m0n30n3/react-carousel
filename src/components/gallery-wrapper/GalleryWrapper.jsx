import React from "react";

import { classNames } from "helpers";
import { TransitionEffect } from "constants";
import { useGalleryTransition } from "hooks";
import { galleryWrapperPropTypes } from "proptypes";

import { base, fade, slide } from "styles/gallery-wrapper.module.css";

export const GalleryWrapper = ({
  imagesLength,
  currentIndex,
  effect,
  effectSpeed,
  ...props
}) => {
  const isFadeEffect = effect === TransitionEffect.FADE;
  const isSlideEffect = effect === TransitionEffect.SLIDE;

  const effects = useGalleryTransition({
    imagesLength,
    currentIndex,
    effectSpeed,
  });

  return (
    <div
      {...props}
      role="listbox"
      className={classNames([
        base,
        isFadeEffect && fade,
        isSlideEffect && slide,
      ])}
      style={effects[effect]}
    />
  );
};

GalleryWrapper.propTypes = galleryWrapperPropTypes;

GalleryWrapper.displayName = "Gallery.GalleryWrapper";
