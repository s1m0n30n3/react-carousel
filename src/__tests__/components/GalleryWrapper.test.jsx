import React from "react";
import renderer from "react-test-renderer";
import { describe, expect, test } from "vitest";

import { TransitionEffect } from "constants";

import { GalleryWrapper } from "components/gallery-wrapper";

describe("GalleryWrapper", () => {
  test("renders correctly without props", () => {
    const component = renderer.create(<GalleryWrapper />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders correctly with props", () => {
    const component = renderer.create(
      <GalleryWrapper
        imagesLength={10}
        currentIndex={1}
        effect={TransitionEffect.FADE}
        effectSpeed={300}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders correctly if `effect` is `slide`", () => {
    const component = renderer.create(
      <GalleryWrapper
        imagesLength={10}
        currentIndex={1}
        effect={TransitionEffect.SLIDE}
        effectSpeed={300}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
