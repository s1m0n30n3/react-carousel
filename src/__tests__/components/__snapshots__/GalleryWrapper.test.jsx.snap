// Vitest Snapshot v1

exports[`GalleryWrapper > renders correctly if \`effect\` is \`slide\` 1`] = `
<div
  className=""
  role="listbox"
  style={
    {
      "transform": "translateX(-10%)",
      "transitionDuration": "300ms",
      "width": "1000%",
    }
  }
/>
`;

exports[`GalleryWrapper > renders correctly with props 1`] = `
<div
  className=""
  role="listbox"
/>
`;

exports[`GalleryWrapper > renders correctly without props 1`] = `
<div
  className=""
  role="listbox"
/>
`;
